src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -340,21 +340,47 @@ trait ContextErrors {
 
       //typedSelect
       def NotAMemberError(sel: Tree, qual: Tree, name: Name) = {
-        def errMsg = {
+        import util.{ EditDistance, StringUtil }
+        def errMsg: String = {
           val owner            = qual.tpe.typeSymbol
           val target           = qual.tpe.widen
           def targetKindString = if (owner.isTypeParameterOrSkolem) "type parameter " else ""
           def nameString       = decodeWithKind(name, owner)
           /* Illuminating some common situations and errors a bit further. */
           def addendum         = {
+            val companionSymbol: Symbol = {
+              if (name.isTermName && owner.isPackageClass)
+                target.member(name.toTypeName)
+              else NoSymbol
+            }
             val companion = {
-              if (name.isTermName && owner.isPackageClass) {
-                target.member(name.toTypeName) match {
-                  case NoSymbol => ""
-                  case sym      => "\nNote: %s exists, but it has no companion object.".format(sym)
-                }
+              if (companionSymbol == NoSymbol) ""
+              else s"\nnote: $companionSymbol exists, but it has no companion object."
+            }
+            // find out all the names available under target within 2 edit distances
+            lazy val alternatives: List[String] = {
+              val editThreshold = 2
+              val x = name.toString
+              if (x.size < 2) Nil
+              else {
+                target.nonPrivateMembers.iterator
```

Comment:
It would be nice to look at all members and filter by those that are accessible at the callsite's `Context`. We should them be careful to exclude fields:

```scala
.filterNot(m => nme.isLocalName(m.name)))
```

Reply:
Added https://github.com/scala/scala/pull/6711/commits/1e8ec90b59f8b8bf44dd58490e5d8f2b4ffdb433 to address the accessibility.